modules/openxr/SCsub

```diff
@@ -32,6 +32,8 @@ elif env["platform"] == "linuxbsd":
 
     if env["x11"]:
         env_openxr.AppendUnique(CPPDEFINES=["XR_USE_PLATFORM_XLIB"])
+    else:
+        env_openxr.Replace(CPPDEFINES=[d for d in env_openxr["CPPDEFINES"] if "GLES3_ENABLED" not in d])
```

Comment:
Hm. Having this in openxr doesn't seem quite right, since X11 being gone implying there's no OpenGL is something that should have a global effect, not just in openxr. (It only happens that it's just the openxr module that's leading to an error in the current code.)

Is there some place higher up in the build system where we could say:

```
if env['platform'] == 'linuxbsd' and not env['x11']:
    env['opengl3'] = False
```

That way anything in the build system that's looking at `env['opengl3']` would be correct. And I think it would lead to the `GLES3_ENABLED` define not being set anywhere, whereas this PR would only be removing that define when compiling openxr.

Reply:
I've moved the content to `SConstruct`, I think it makes more sense. Please let me know if there is maybe a better option :)

Reply:
This looks good to me!

@akien-mga (as maintainer of the build system) will be the final judge :-)